test/test_base.py

```diff
@@ -13,71 +13,87 @@
 
 class TestBase:
     def __init__(self):
-        # Warnet server stdout gets logged here
-        self.tmpdir = Path(mkdtemp(prefix="warnet-test-"))
-        os.environ["XDG_STATE_HOME"] = f"{self.tmpdir}"
-        self.logfilepath = self.tmpdir / "warnet" / "warnet.log"
+        self.setup_environment()
+        self.setup_logging()
+        atexit.register(self.cleanup)
+        self.log.info("Warnet test base initialized")
 
+    def setup_environment(self):
+        self.tmpdir = Path(mkdtemp(prefix="warnet-test-"))
+        os.environ["XDG_STATE_HOME"] = str(self.tmpdir)
+        self.logfilepath = self.tmpdir / "warnet.log"
         # Use the same dir name for the warnet network name
         # replacing underscores which throws off k8s
         self.network_name = self.tmpdir.name.replace("_", "")
-
         self.server = None
         self.server_thread = None
         self.stop_threads = threading.Event()
         self.network = True
 
-        atexit.register(self.cleanup)
-
-        print("\nWarnet test base started")
+    def setup_logging(self):
+        import logging
+
+        self.log = logging.getLogger("TestFramework")
+        self.log.setLevel(logging.DEBUG)
+        fh = logging.FileHandler(self.logfilepath, encoding="utf-8")
+        fh.setLevel(logging.DEBUG)
+        ch = logging.StreamHandler()
+        ch.setLevel(logging.DEBUG)
+        formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
```

Comment:
something is a bit funky here, im seeing double timestamps in some but not all log messages:

```

2024-07-08 15:18:43,320 - TestFramework - DEBUG - Node 2 has 1 settled invoices
2024-07-08 15:18:48,326 - TestFramework - DEBUG - Executing warcli command: lncli 2 listinvoices
2024-07-08 15:18:48,711 - TestFramework - DEBUG - 2024-07-08T19:18:48+0000 - DEBUG - {'jsonrpc': '2.0', 'method': 'tank_lncli', 'params': {'network': 'warnet-test-wspv4wt2', 'node': 2, 'command': ['listinvoices']}, 'id': 1}
2024-07-08 15:18:48,714 - TestFramework - DEBUG - 2024-07-08T19:18:48+0000 - DEBUG - Running lncli cmd=lncli --network=regtest listinvoices on tank.index=2
2024-07-08 15:18:48,714 - TestFramework - DEBUG - 2024-07-08T19:18:48+0000 - DEBUG - Running exec_cmd=['/bin/sh', '-c', 'lncli --network=regtest listinvoices'] on tank_index=2
```

Reply:
Yeah, i need to update this to use our logging.config too: https://github.com/bitcoin-dev-project/warnet/pull/384#issuecomment-2214747548

Reply:
fixed in e1373a91bd7937de1d0dd5f7d746301e76b2c5d8